.github/workflows/nightly-build.yaml

```
@@ -0,0 +1,76 @@
+name: Nightly builds
+
+on:
+  # workflow_dispatch so that it can be triggered manually if needed
+  workflow_dispatch:
+
+  schedule:
+    - cron: "0 0 * * *"
```

There is most probably a peak of cron executions at this time. I would use another random time, for example:

```suggestion
    - cron: "34 21 * * *"
```